### PR TITLE
Existance Tests datastore collection and tests reorganised for pytest

### PR DESCRIPTION
**Pull request by jyejare**
This PR includes:

1. The datastore set for preupgrade entites and post upgrade entities for existance tests
2. The test_existance file is deleted that was having tests for sll the entities where there was problem with running those tests with py.test(for subtest module from unittest and tests inside classes) and with unittest2 generating report at the end of all tests.
3. So, I decided to separate each entity related in its own module. So that we can run tests with py.test and generate reports which can be send with email notification.

### CODE REVIEW

**Comment by jyejare** [comment]
py.test logs:

```
$ py.test upgrade_tests/test_existance_relations/
================================================================================================ test session starts ================================================================================================
platform linux2 -- Python 2.7.10, pytest-2.8.7, py-1.4.31, pluggy-0.3.1
rootdir: /home/jitendrayejare2/Desktop/RedHat/ToolsTop/automation-tools, inifile:
plugins: xdist-1.15.0, cov-2.4.0
collected 48 items

upgrade_tests/test_existance_relations/test_activationkeys.py ........
upgrade_tests/test_existance_relations/test_capsules.py FF..
upgrade_tests/test_existance_relations/test_contentviews.py ............
upgrade_tests/test_existance_relations/test_hosts.py ....F........F........
upgrade_tests/test_existance_relations/test_organizations.py ..

===================================================================================================== FAILURES ======================================================================================================
__ test_positive_capsules_by_features[tftp, dns, dhcp, puppet, puppet ca, bmc, pulp node, templates, discovery, openscap-tftp, http, dns, dhcp, puppet, puppet ca, bmc, pulp node, templates, discovery, openscap] __

pre = 'tftp, dns, dhcp, puppet, puppet ca, bmc, pulp node, templates, discovery, openscap', post = 'tftp, http, dns, dhcp, puppet, puppet ca, bmc, pulp node, templates, discovery, openscap'

    @pytest.mark.parametrize(
        "pre,post",
        compare_postupgrade('capsule', 'features')
    )
    def test_positive_capsules_by_features(pre, post):
        """Test all features of each capsule are existing post upgrade

        :id: 6d3b8f24-2d51-465c-8d01-5a159aa89f2f

        :assert: All features of all capsules should be retained post upgrade
        """
>       assert pre == post
E       assert 'tftp, dns, d...ery, openscap' == 'tftp, http, d...ery, openscap'
E         - tftp, dns, dhcp, puppet, puppet ca, bmc, pulp node, templates, discovery, openscap
E         + tftp, http, dns, dhcp, puppet, puppet ca, bmc, pulp node, templates, discovery, openscap
E         ?       ++++++

upgrade_tests/test_existance_relations/test_capsules.py:21: AssertionError
______________ test_positive_capsules_by_features[tftp, dns, dhcp, puppet, puppet ca, bmc, pulp, discovery, openscap, dynflow, ssh-tftp, dns, dhcp, puppet, puppet ca, pulp, discovery, dynflow, ssh] _______________

pre = 'tftp, dns, dhcp, puppet, puppet ca, bmc, pulp, discovery, openscap, dynflow, ssh', post = 'tftp, dns, dhcp, puppet, puppet ca, pulp, discovery, dynflow, ssh'

    @pytest.mark.parametrize(
        "pre,post",
        compare_postupgrade('capsule', 'features')
    )
    def test_positive_capsules_by_features(pre, post):
        """Test all features of each capsule are existing post upgrade

        :id: 6d3b8f24-2d51-465c-8d01-5a159aa89f2f

        :assert: All features of all capsules should be retained post upgrade
        """
>       assert pre == post
E       assert 'tftp, dns, d... dynflow, ssh' == 'tftp, dns, dh... dynflow, ssh'
E         - tftp, dns, dhcp, puppet, puppet ca, bmc, pulp, discovery, openscap, dynflow, ssh
E         ?                                    -----                  ----------
E         + tftp, dns, dhcp, puppet, puppet ca, pulp, discovery, dynflow, ssh

upgrade_tests/test_existance_relations/test_capsules.py:21: AssertionError
_______________________________________________________________________________ test_positive_hosts_by_ip[410.8.119.134-10.8.119.100] _______________________________________________________________________________

pre = '10.8.119.134', post = '10.8.119.100'

    @pytest.mark.parametrize("pre,post", compare_postupgrade('host', 'ip'))
    def test_positive_hosts_by_ip(pre, post):
        """Test ip associations of all hosts post upgrade

        :id: 3b4f8315-8490-42bc-8afa-4a6c267558d7

        :assert: IP of each host should be associated to its respective host
            post upgrade
        """
>       assert pre == post
E       assert '10.8.119.134' == '10.8.119.100'
E         - 10.8.119.134
E         ?           ^^
E         + 10.8.119.100
E         ?           ^^

upgrade_tests/test_existance_relations/test_hosts.py:19: AssertionError
_________________________________________________________________________ test_positive_hosts_by_mac[2ce:70:5c:40:96:74-ce:70:5c:40:96:e4] __________________________________________________________________________

pre = 'ce:70:5c:40:96:74', post = 'ce:70:5c:40:96:e4'

    @pytest.mark.parametrize("pre,post", compare_postupgrade('host', 'mac'))
    def test_positive_hosts_by_mac(pre, post):
        """Test mac associations of all hosts post upgrade

        :id: 526af1dd-f2a1-4a66-a0d2-fe5c1ade165d

        :assert: MAC of each host should be associated to its respective host
            post upgrade
        """
>       assert pre == post
E       assert 'ce:70:5c:40:96:74' == 'ce:70:5c:40:96:e4'
E         - ce:70:5c:40:96:74
E         ?                ^
E         + ce:70:5c:40:96:e4
E         ?                ^

upgrade_tests/test_existance_relations/test_hosts.py:31: AssertionError
======================================================================================== 4 failed, 44 passed in 0.62 seconds ========================================================================================
```

**Comment by sghai** [comment]
@jyejare : that's how I was expecting the test results.. Looks good now.. and I feel that  having separate tests in separate modules would be easier to manage..

But looks like time to have separate upgrade test suite :)

**Comment by jyejare** [comment]
@sghai, surely with the increasing size of tests and modules and what @san7ket is working on, definitely we need to separate upgrade from automation-tools.

File: automation_tools/satellite6/upgrade/satellite.py [hunk 1]
```diff
@@ -245,3 +252,6 @@ def satellite6_zstream_upgrade():
     set_hammer_config()
     hammer('ping')
     run('katello-service status', warn_only=True)
+    if os.environ.get('RUN_EXISTANCE_TESTS'):
```
Comment (sghai) [inline automation_tools/satellite6/upgrade/satellite.py]: are we setting this variable for trigger too ?

Comment (jyejare) [inline automation_tools/satellite6/upgrade/satellite.py]: Yes once we introduce tests in trigger job, we can have to add this var.

File: upgrade_tests/test_existance_relations/test_capsules.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"""Upgrade TestSuite for validating Satellite capsules existence and
+associations post upgrade
+"""
+from automation_tools.satellite6.upgrade.tests.existence import (
+    compare_postupgrade
+)
+import pytest
+
```
Comment (sghai) [inline upgrade_tests/test_existance_relations/test_capsules.py]: same as above

**Comment by sghai** [comment]
ACK.. PR looks good.. I really like the test results..

**Pull request merged**